### PR TITLE
added bias_Add

### LINKED ISSUE #21483: bias_add

### PR DESCRIPTION
**Pull request by sushmanthreddy**
close #21483

### CODE REVIEW

**Review by hello-fri-end — COMMENTED** [review]
Is this https://mindspore.cn/docs/en/r2.0/api_python/ops/mindspore.ops.bias_add.html the function you're implementing?  If yes, the function signature is not matching and if not, could you please drop a link of the mindspore `bias_add` function here? Thanks for looking into this :)

**Comment by sushmanthreddy** [comment]
@hello-fri-end there is problem here in this mindspore biasadd...

there are two links for this implementation

[link1](https://mindspore.cn/docs/en/r2.0/api_python/ops/mindspore.ops.BiasAdd.html?highlight=bias_add) and another implementation of bias_add [link2](https://mindspore.cn/docs/en/r2.0/api_python/ops/mindspore.ops.bias_add.html?highlight=bias_add#mindspore.ops.bias_add)


so I have combined both of them..
if u need only one of the implementation plss let me knew ,which I need to implement link1 or link2

**Comment by hello-fri-end** [comment]
> @hello-fri-end there is problem here in this mindspore biasadd...
> 
> there are two links for this implementation
> 
> [link1](https://mindspore.cn/docs/en/r2.0/api_python/ops/mindspore.ops.BiasAdd.html?highlight=bias_add) and another implementation of bias_add [link2](https://mindspore.cn/docs/en/r2.0/api_python/ops/mindspore.ops.bias_add.html?highlight=bias_add#mindspore.ops.bias_add)
> 
> so I have combined both of them.. if u need only one of the implementation plss let me knew ,which I need to implement link1 or link2

Hey @sushmanthreddy ! You should be implementing the function in the second link i.e `mindspore.ops.bias_add` function. Would you please update your implementation as per the native mindspore function? Feel free to request another review when you are done with the changes. Thanks.

**Comment by sushmanthreddy** [comment]
@hello-fri-end I have added the test function and main function as u mentioned ..
can u pls review it and give feedback

File: ivy/functional/frontends/mindspore/ops/function/nn_func.py [hunk 1]
```diff
@@ -372,6 +383,22 @@ def avg_pool2d(
     )
 
 
+@to_ivy_arrays_and_back
```
Comment (hello-fri-end) [inline ivy/functional/frontends/mindspore/ops/function/nn_func.py:386]: Could you please use the following script to determine the data-types supported by this function and add the supported or unsupported dtypes to `with_supported_dtypes` or `with_unsupported_dtypes` decorator please?
```python
import mindspore.ops as ops
import mindspore.common.dtype as mstype

def check_bias_add_supported_dtypes():
    supported_dtypes = [
        mstype.float16, mstype.float32, mstype.float64,
        mstype.int8, mstype.int16, mstype.int32, mstype.int64,
        mstype.uint8, mstype.uint16
    ]
    unsupported_dtypes = []

    for dtype in supported_dtypes:
        try:
            _ = ops.bias_add([1, 2, 3], [0.1, 0.2, 0.3], data_type=dtype)
        except ValueError:
            unsupported_dtypes.append(dtype)

    return supported_dtypes, unsupported_dtypes

if __name__ == "__main__":
    supported_dtypes, unsupported_dtypes = check_bias_add_supported_dtypes()
    
    print("Supported Data Types:")
    for dtype in supported_dtypes:
        print(dtype)
    
    print("\nUnsupported Data Types:")
    for dtype in unsupported_dtypes:
        print(dtype)
```

File: ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py [hunk 2]
```diff
@@ -428,6 +428,39 @@
...
+#     helpers.test_frontend_function(
+#         input_dtypes=dtype,
+#         frontend=frontend,
+#         test_flags=test_flags,
+#         fn_tree=fn_tree,
+#         on_device=on_device,
+#         input=x[0],
+#         bias=bias,
+#     )
+
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py:463]: Why is this commented out?

**Review by hello-fri-end — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/mindspore/ops/function/nn_func.py (hunk 1 above)
Comment (sushmanthreddy) [reply ivy/functional/frontends/mindspore/ops/function/nn_func.py:386]: @hello-fri-end i really dont knew how to run that script,i really dont knew where should i add for getting those suggestions..can u pls tell me how to run that?

File: ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py (hunk 2 above)
Comment (sushmanthreddy) [reply ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py:463]: the mindspore support is not there yet and testing pipeline,so we are commenting out ..

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**